repl/evaluator.go

```diff
@@ -586,15 +652,104 @@ func (e *Evaluator) applyContext() (*cel.Env, interpreter.Activation, error) {
 		return nil, nil, err
 	}
 
-	env := e.env
+	return e.ctx.getEffectiveEnv(e.env), act, nil
+}
+
+// typeOption implements optioner for loading a set of types defined by a protobuf file descriptor set.
+type typeOption struct {
+	path string
+	fds  *descpb.FileDescriptorSet
+}
+
+func (o *typeOption) String() string {
+	return fmt.Sprintf("%%load_descriptors '%s'", o.path)
+}
+
+func (o *typeOption) Option() cel.EnvOption {
+	return cel.TypeDescs(o.fds)
+}
+
+type containerOption struct {
+	container string
+}
+
+func (o *containerOption) String() string {
+	return fmt.Sprintf("%%option --container '%s'", o.container)
+}
+
+func (o *containerOption) Option() cel.EnvOption {
+	return cel.Container(o.container)
+}
+
+// setOption sets a number of options on the environment. returns an error if
+// any of them fail.
+func (e *Evaluator) setOption(args []string) error {
+	var issues []string
+	for idx := 0; idx < len(args); {
+		arg := args[idx]
+		idx++
+		if arg == "--container" {
+			if idx >= len(args) {
+				issues = append(issues, "not enough args for container")
+			}
+			container := args[idx]
+			idx++
+			err := e.AddOption(&containerOption{container: container})
+			if err != nil {
+				issues = append(issues, fmt.Sprintf("container: %v", err))
+			}
+		} else {
+			issues = append(issues, fmt.Sprintf("unsupported option '%s'", arg))
+		}
+	}
+	if len(issues) > 0 {
+		return errors.New(strings.Join(issues, "\n"))
+	}
+	return nil
+}
+
+func loadFileDescriptorSet(path string, textfmt bool) (*descpb.FileDescriptorSet, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var fds descpb.FileDescriptorSet
 
-	if len(e.ctx.letVars) > 0 {
-		env = e.ctx.letVars[len(e.ctx.letVars)-1].env
-	} else if len(e.ctx.letFns) > 0 {
-		env = e.ctx.letFns[len(e.ctx.letFns)-1].env
+	if textfmt {
+		err = prototext.Unmarshal(data, &fds)
+	} else {
+		// binary pb
+		err = proto.Unmarshal(data, &fds)
+	}
+	if err != nil {
+		return nil, err
 	}
 
-	return env, act, nil
+	return &fds, nil
+}
+
+func (e *Evaluator) loadDescriptors(args []string) error {
+	if len(args) < 1 {
+		return errors.New("expected path for load descriptors")
+	}
+	flag := ""
+	if len(args) > 1 {
+		flag = args[0]
+	}
+	textfmt := true
+	if flag == "--textproto" {
```

Comment:
You can probably omit this flag check since the default will load the protos in `textFmt`. Instead just use the remaining `if flag == "--binarypb" {` to switch to reading binary format.